From b3209e232fb09fd95c74d57b8d28e5ca72d3de34 Mon Sep 17 00:00:00 2001
From: Dmitry Myaskovskiy <dmitry@enapter.com>
Date: Wed, 27 Jul 2022 19:18:20 +0300
Subject: [PATCH] Replace login

Upstream-Status: Inappropriate [Enapter specific]
---
 units/getty@.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/getty@.service.in b/units/getty@.service.in
index 21d66f9367..51370ccf09 100644
--- a/units/getty@.service.in
+++ b/units/getty@.service.in
@@ -38,7 +38,7 @@ ConditionPathExists=/dev/tty0
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
+ExecStart=-/sbin/agetty --skip-login --login-program=/usr/sbin/enapter-login - $TERM
 Type=idle
 Restart=always
 RestartSec=0
-- 
2.36.1

